### DISCUSSION TITLE
Is the client builders for AWS s3 or any other service side effecting?

### DISCUSSION THREAD

**Discussion by VladKopanievITV** [aws/aws-sdk-java#2949] [Q&A] [answered]
Would the code like this  `AmazonS3ClientBuilder.standard().build()` execute any IO, open connections or change global mutable state?

**Comment by debora-ito** [comment, marked as answer]
The instance of S3 client returned by `build()` is immutable.
It will create the connection pool, but not open connections at startup. Connections will be created lazily.

**Reply by VladKopanievITV** [reply]
Nice, thanks for confirming, Debora!
